tests/test.py

```diff
@@ -0,0 +1,100 @@
+import glob
+import os
+import re
+import subprocess
+import sys
+import time
+import shutil
+import socket
+
+from avocado import VERSION
+from avocado import Test
+
+class MinishiftISOTest(Test):
+
+    def setUp(self):
+        ''' Test Setup '''
+        self.log.info("################################################################")
+        self.log.info("Avocado version : %s" % VERSION)
+        self.log.info("################################################################")
+
+        self.repo_dir = os.path.dirname(os.path.realpath(__file__)) + "/.."
+        self.scripts_dir = self.repo_dir + "/tests/"
+        self.bin_dir = self.repo_dir + "/build/bin/"
+        self.driver_name = 'kvm'
+        self.iso_name = 'minishift-b2d'
+
+        # Find iso files and skip test if no ISO file is found
+        self.iso_file = self.repo_dir + "/build/%s.iso" % self.iso_name
+        if not os.path.isfile(self.iso_file):
+            self.skip("Skipping testing as no ISO found in 'build' directory.")
+
+    def test_boot_vm_out_of_iso(self):
+        ''' Test booting up VM out of ISO '''
+        start_args = (self.driver_name, "file://"  + self.iso_file)
+        cmd = self.bin_dir + "minishift start --vm-driver %s --iso-url %s" % start_args
+        self.execute_test({ 'cmd': cmd })
+
+    def test_ssh_connection_to_vm(self):
+        ''' Test SSH connection to the VM '''
+        cmd = self.bin_dir + "minishift ssh exit"
+        self.execute_test({ 'cmd': cmd })
+
+    def test_ip_of_vm(self):
+        ''' Test IP of the VM '''
+        cmd = self.bin_dir + "minishift ip"
+        output = self.execute_test({ 'cmd': cmd })
+
+        # Verify IP
+        try:
+            socket.inet_aton(output)
+        except socket.error:
+            self.fail("Error in getting IP with minishift VM.")
+
+    def test_docker_env_evaluable(self):
+        cmd = self.bin_dir + "minishift docker-env"
+        output = self.execute_test({ 'cmd': cmd })
+        self.check_data_evaluable(output.rstrip())
+
+    def test_stopping_vm(self):
+        ''' Test stopping machine '''
+        cmd = self.bin_dir + "minishift stop"
+        self.execute_test({ 'cmd': cmd })
+
+        # Check VM status status
+        cmd = self.bin_dir + "minishift status"
+        self.log.info("Executing command : %s" % cmd)
+        process = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+        output = process.communicate()[0]
+        self.assertEqual(0, process.returncode)
+        self.assertEqual('Stopped', output.rstrip())
+
+    def test_removing_vm(self):
```

Comment:
instead of remove I would say let's call it `test_delete_vm` just to align with our methods.